portal/src/main/scala/cromwell/pipeline/service/UserService.scala

```diff
@@ -0,0 +1,54 @@
+package cromwell.pipeline.service
+
+import akka.http.scaladsl.model.StatusCodes
+import akka.http.scaladsl.server.Directives.complete
+import akka.http.scaladsl.server.Route
+import cromwell.pipeline.datastorage.dao.repository.UserRepository
+import cromwell.pipeline.datastorage.dto.user.UpdateRequest
+import cromwell.pipeline.datastorage.dto.user.{ UpdatePasswordRequest, UpdateRequest }
+import cromwell.pipeline.datastorage.dto.{ User, UserId }
+import cromwell.pipeline.utils.StringUtils
+import play.api.libs.json.JsError
+import play.api.libs.json.JsResult.Exception
+
+import scala.concurrent.{ ExecutionContext, Future }
+import scala.util.Random
+
+class UserService(userRepository: UserRepository)(implicit executionContext: ExecutionContext) {
+
+  def updateUser(id: String, request: UpdateRequest): Future[Int] = {
+    val updatedUser = User(
+      userId = UserId(id),
+      email = request.email,
+      passwordSalt = "",
+      passwordHash = "",
+      firstName = request.firstName,
+      lastName = request.lastName
+    )
+    userRepository.updateUser(updatedUser)
+  }
+
+  def updateUserPassword(id: String, request: UpdatePasswordRequest): Future[Int] =
+    if (request.newPassword != request.repeatPassword) {
+      throw Exception(JsError("different password"))
+    } else {
+      val passwordSalt = getSalt
+      val passwordHash = getPasswordHash(request.currentPassword, passwordSalt)
+      val updatedUser = User(
+        userId = UserId(id),
+        email = "",
+        passwordSalt = passwordSalt,
+        passwordHash = passwordHash,
+        firstName = "",
+        lastName = ""
+      )
+      userRepository.updateUserPassword(updatedUser)
+    }
+
+  def getStatus(count: Int): Route =
```

Comment:
This belongs to the `Controller` layer, not service.

Reply:
I would argue that the `Controller` should not decide whether it is good or bad that we have updated exactly one user. In my opinion, `Service` layer should have return `Success` _only_ if it really succeeded (i.e. updated only one user) and `Failed` otherwise.

Reply:
Done. Chose Nikita's variant